### DISCUSSION TITLE
How to provide stylings to preferences items.

### DISCUSSION THREAD

**Discussion by Abhay0196** [eclipse-theia/theia#13398] [Q&A] [answered]
Hi,

I have my custom preference category(say ABC) and inside which I have a sub category(say XYZ). Under this I have a preference option for which I have attached a snap. In snap it is pointed that checkbox and the sentence are not aligned properly.

<img width="440" alt="image" src="https://github.com/eclipse-theia/theia/assets/158473054/67005167-70ef-4ccb-a4c8-411ea8381e07">

Below is the code for adding this preference item:
```ts
export const xyzSchemaGenerator = (extensionName: string) => ({
    type: 'object',
    properties: {
        [`ABC.XYZ.${extensionName}.sampleSubCategory`]: {
            description: `Sample sub category of ${extensionName}`,
            additionalProperties: {
                type: 'string'
            }
        }
    }
});
```

It is properly binded in the frontend-module.ts file.

FYI, this behavior is same for default properties as well.

Requesting you to provide suggestion on how it be aligned properly or is there anything missing the given code.

**Comment by msujew** [comment, marked as answer]
Hey @Abhay0196,

you're missing the `type` property. See:

https://github.com/eclipse-theia/theia/blob/0cb16bc61661001f753723f377fe4c9639ed35aa/packages/core/src/browser/core-preferences.ts#L58-L63

I believe TypeScript would correctly complain about it missing if you add the `PreferenceSchema` type as the return type of your function.

If that doesn't help, you can always just import a custom `.css` file in your code which targets the HTML elements in Theia. That way you can write any CSS to align your checkbox how you like.

**Reply by Abhay0196** [reply]
Thanks a lot @msujew for the response. It worked.
